pentaho/src/main/java/pt/webdetails/cgg/CggService.java

```diff
@@ -211,6 +214,11 @@ public void setResponseHeader( String mimeType ) {
 
     } catch ( Exception ex ) {
       logger.fatal( "Error while rendering script", ex );
+      if (ex.getMessage().contains("ScriptResourceNotFoundException")){
+        servletResponse.setStatus( 404 );
+      } else {
+        servletResponse.setStatus( 400 );
+      }
```

Comment:
Why not look at the getCause() of `ex` in search for a nested `ScriptResourceNotFoundException`? Deciding from the exception text would be a last resort approach.